### PR TITLE
LWDEV-9284 Add swagger method to turn on/off maintenance mode

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/holders/MessageProcessingSwitchStatusHolder.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.lykke.matching.engine.holders
+
+import com.lykke.matching.engine.database.cache.ApplicationSettingsCache
+import com.lykke.matching.engine.utils.monitoring.GeneralHealthMonitor
+import org.springframework.stereotype.Component
+
+@Component
+class MessageProcessingStatusHolder(private val generalHealthMonitor: GeneralHealthMonitor,
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/holders/MessageProcessingSwitchStatusHolder.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/events/SettingChangedEvent.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.lykke.matching.engine.services.events
+
+import com.lykke.matching.engine.daos.setting.AvailableSettingGroup
+import com.lykke.matching.engine.daos.setting.Setting
+
+class SettingChangedEvent(val settingGroup: AvailableSettingGroup,
+                          val settingName: String,
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/events/SettingChangedEvent.kt]: removed

File: src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt [hunk 3]
```diff
@@ -53,7 +53,6 @@ class MessageStatusUtils {
                 ValidationException.Validation.BALANCE_LOWER_THAN_RESERVED -> MessageStatus.BALANCE_LOWER_THAN_RESERVED
                 ValidationException.Validation.NEGATIVE_OVERDRAFT_LIMIT -> MessageStatus.NEGATIVE_OVERDRAFT_LIMIT
                 ValidationException.Validation.LIMIT_ORDER_NOT_FOUND -> MessageStatus.LIMIT_ORDER_NOT_FOUND
-                ValidationException.Validation.NEGATIVE_OVERDRAFT_LIMIT -> MessageStatus.NEGATIVE_OVERDRAFT_LIMIT
                 ValidationException.Validation.NOT_ACCEPTABLE_MESSAGE_SWITCH_SETTING_VALUE -> MessageStatus.BAD_REQUEST
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: Remove this Validation value from here and from enum

Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: removed - was added during merge to dev branch

**Pull request merged**